### PR TITLE
fix(facets): update hierarchical facet model and replace adapter

### PR DESCRIPTION
**Pull request by pmareke**
EX-4135

The response from the new API changes a couple of things:

- The **hierarchical category** now it's **categoryPaths**
- The children values are now under a new property called **children**

NOW
image
BEFORE
image

In order to achieve that change we need to replace on of the mappers of the facets
image

The **EmpathyHierarchicalFacetMapper** is the one to override, after creating the new mapper we need to do the replacement and the only way I found is the following
image

### CODE REVIEW

**Review by davidmfempathy — DISMISSED** [review]

File: src/components/facets/facets.vue [hunk 1]
```diff
@@ -11,6 +11,32 @@
...
+            :filter="hierarchicalFilter"
+          >
+            <label :class="cssClasses">
+              <input @change="clickFilter" :disabled="isDisabled" type="checkbox" />
+              {{ filter.label }}
+            </label>
+          </HierarchicalFilter>
+        </Filters>
+      </BaseHeaderTogglePanel>
+    </template>
```
Comment (tajespasarela) [inline src/components/facets/facets.vue:38]: change this to use the current Facet component as in the rest of the Facets.
Also remove any custom content in the filter, this is out of the scope of this task:

```html
    <template #categoryPaths="{ facet }">
      <Facet :facet="facet">
        <AllFilter :facet="facet" />
        <Filters #default="{ filter }" :filters="facet.filters" :animation="staggeredFadeAndSlide">
          <HierarchicalFilter :filter="filter" />
        </Filters>
      </Facet>
    </template>
```

**Review by tajespasarela — CHANGES_REQUESTED** [review]
remove the previous `q` parameter in the request: 
```
export function customRequestMapper(
  rawRequest: SearchRequest,
  request: EmpathySearchRequest
): EmpathySearchRequest {
  return Object.assign(request, {
    query: rawRequest.query,
    q: undefined
  });
}
```

File: src/x-components/adapter.ts [hunk 2]
```diff
@@ -13,5 +20,13 @@ export const adapter = new EmpathyAdapterBuilder()
...
+  .setFacetConfig({ modelName: 'HierarchicalFacet' }, 'categoryPaths')
+  .configureContainer(container => {
+    container.unbind(DEPENDENCIES.ResponseMappers.facets);
+    container.bind(DEPENDENCIES.ResponseMappers.facets).to(EmpathyFacetMapper);
+    container.bind(DEPENDENCIES.ResponseMappers.facets).to(HierarchicalFacetMapper);
+    container.bind(DEPENDENCIES.ResponseMappers.facets).to(EmpathyNumberRangeFacetMapper);
+    container.bind(DEPENDENCIES.ResponseMappers.facets).to(EmpathySimpleFacetMapper);
+  })
   .setInstance('platform')
   .build();
```
Comment (tajespasarela) [inline src/x-components/adapter.ts:32]: Move this file to `src/adapters` and rename the folder to  `src/adapter`

**Review by tajespasarela — CHANGES_REQUESTED** [review]

**Comment by pmareke** [comment]
> q: undefined

Nice, I tried to remove it but I didn't find the way. Thanks!

File: src/x-components/adapter.ts [hunk 3]
```diff
@@ -1,29 +0,0 @@
...
-    responsePaths: {
-      relatedTags: 'data.relatedtags'
-    }
-  })
-  .setFeatureConfig('nextQueries', {
-    endpoint: 'https://search.internal.test.empathy.co/query/empathy/nextqueries',
-    responsePaths: {
-      nextQueries: 'data.nextQueries'
-    }
-  })
```
Comment (tajespasarela) [inline src/x-components/adapter.ts:27]: undo the deletion of the NQ and RT configs

**Review by tajespasarela — CHANGES_REQUESTED** [review]

File: src/components/facets/facets.vue (hunk 1 above)
Comment (tajespasarela) [reply src/components/facets/facets.vue:38]: where is this change?

**Review by tajespasarela — CHANGES_REQUESTED** [review]
There is a pending requested change in `facets.vue`

File: src/components/facets/facets.vue (hunk 1 above)
Comment (pmareke) [reply src/components/facets/facets.vue:38]: I did it, I promise...

**Pull request merged**